Code/Framework/AzToolsFramework/Tests/Prefab/PrefabDeleteAsOverrideTests.cpp

```diff
@@ -279,4 +279,90 @@ namespace UnitTest
+                {
+                    wheelInstanceContainerId = nestedInstance->GetContainerEntityId();
+                    return;
+                }
+            });
+        ASSERT_TRUE(wheelInstanceContainerId.IsValid()) << "Cannot get wheel container entity id in the car.";
+
+        // Delete the Wheel instance. This adds an override on the level/root prefab
+        PrefabOperationResult result = m_prefabPublicInterface->DeleteEntitiesAndAllDescendantsInInstance({ wheelInstanceContainerId });
+        ASSERT_TRUE(result.IsSuccess());
+
+        // Propagate changes after deleting the Wheel instance
+        ProcessDeferredUpdates();
+
+        // Validate that the Wheel instance is not under the Car instance
+        ValidateNestedInstanceNotUnderInstance(carContainerId, wheelInstanceAlias);
+
+        // Focus on the Car instance
+        auto* prefabFocusPublicInterface = AZ::Interface<PrefabFocusPublicInterface>::Get();
+        EXPECT_TRUE(prefabFocusPublicInterface != nullptr);
+        PrefabFocusOperationResult focusResult = prefabFocusPublicInterface->FocusOnOwningPrefab(carContainerId);
+        EXPECT_TRUE(focusResult.IsSuccess());
+
+        // Propagate changes after the focus change
+        ProcessDeferredUpdates();
+
+        // Find the focused Car instance
+        AzFramework::EntityContextId editorEntityContextId = AzToolsFramework::GetEntityContextId();
+        AZ::EntityId focusedCarContainerId = prefabFocusPublicInterface->GetFocusedPrefabContainerEntityId(editorEntityContextId);
+        ASSERT_TRUE(focusedCarContainerId.IsValid()) << "Cannot get the focused instance.";
+
+        // Validate that the Wheel instance exists again
+        InstanceAlias wheelInstanceAliasInFocusedCar = FindNestedInstanceAliasInInstance(focusedCarContainerId, wheelPrefabName);
+        EXPECT_TRUE(!wheelInstanceAliasInFocusedCar.empty());
+
+        // Find the Wheel instance under the focused Car instance
+        AZ::EntityId wheelInstanceContainerIdInFocusedCar;
+        auto focusedCarInstance = m_instanceEntityMapperInterface->FindOwningInstance(focusedCarContainerId);
+        focusedCarInstance->get().GetNestedInstances(
+            [&wheelInstanceContainerIdInFocusedCar, wheelInstanceAliasInFocusedCar](AZStd::unique_ptr<Instance>& nestedInstance)
+            {
+                if (nestedInstance->GetInstanceAlias() == wheelInstanceAliasInFocusedCar)
+                {
+                    wheelInstanceContainerIdInFocusedCar = nestedInstance->GetContainerEntityId();
+                    return;
+                }
+            });
+        ASSERT_TRUE(wheelInstanceContainerIdInFocusedCar.IsValid()) << "Cannot get wheel container entity id after focusing on Car.";
+
+        // Focus on the Wheel instance
+        focusResult = prefabFocusPublicInterface->FocusOnOwningPrefab(wheelInstanceContainerIdInFocusedCar);
+        EXPECT_TRUE(focusResult.IsSuccess());
```

Comment:
In addition to this, should we verify the transform (parent entity) of the wheel container entity?

Reply:
Yeah, it would be good to check that the DOMs are being retrieved from the correct ancestor, etc. I spun off a separate ticket for adding more unit tests. I'll see if I can add a simple check here too.